numpy/_core/tests/test_umath.py

```diff
@@ -1235,6 +1235,16 @@ def test_float_to_inf_power(self):
             r = np.array([1, 1, np.inf, 0, np.inf, 0, np.inf, 0], dt)
             assert_equal(np.power(a, b), r)
 
+    def test_power_fast_paths(self):
+        # gh-26055
+        for dt in [np.float32, np.float64]:
+            expected = np.array([0.0, 1.21, 4., 1.44e+26, np.inf, np.inf])
+            a = np.array([0, 1.1, 2, 12e12, np.inf, -np.inf], dt)
+            assert_equal(np.power(a, 2.), expected.astype(dt))
```

Comment:
`assert_equal` it too strict a condition. For example, 1.21 cannot be represented in float32 and is actually stored as 1.21000003814697265625, which will differ from the output of `1.1*1.1`. I would recommend using `assert_array_max_ulp` with maxulp = 1.